ladders.js

```diff
@@ -38,6 +38,28 @@ class BattleReady {
 		/** @type {number} */
 		this.time = Date.now();
 	}
+
+	/**
+	 * 
+	 * @param {BattleReady} first The first item to compare
+	 * @param {BattleReady} second The second item to compare
+	 */
+	static sameFormatAs(first, second) { 
+		return first.formatid === second.formatid;
+	}
+
+	/**
+	 * 
+	 * @param {[BattleReady]} battleReadies	An array of battle readies to validate if they are playing the same game. 
+	 */
+	static validatePlayingSameGame(battleReadies) {
+		const firstItem = battleReadies[0];
+		const allSameFormat = battleReadies.every((battleReady) => {
+			return BattleReady.sameFormatAs(firstItem, battleReady); 
+		});
```

Comment:
As this is an inner loop, I would use high-performance code rather than `.every`. This doesn't look like it even saves much readability over the for-loop; you're still extracting out `firstItem` manually.